src/index.js

```diff
@@ -1,29 +1,73 @@
+/* eslint-disable no-use-before-define */
 import './styles.css';
+import { addNewTask, form, input } from './addNewTask.js';
+import removeTask from './removeTask.js';
 
 const placeholder = document.querySelector('ul');
 
-const simpleTodoTasks = [
-  {
-    description: 'Wash the dishes',
-    completed: false,
-    index: 0,
-  },
-  {
-    description: 'complete To Do list project',
-    completed: false,
-    index: 0,
-  },
-];
+const simpleTodoTasks = JSON.parse(localStorage.getItem('task')) || [];
 
 const populateEachTask = (arr) => {
+  localStorage.setItem('task', JSON.stringify(simpleTodoTasks));
+  placeholder.innerHTML = '';
   for (let i = 0; i < arr.length; i += 1) {
     const taskDetails = arr[i];
     const taskContainer = document.createElement('li');
+    taskContainer.setAttribute('data-id', i);
     taskContainer.className = 'task-container';
-    taskContainer.innerHTML = `<span class='task-content'><input type='checkbox'></span><span class='task-content'>${taskDetails.description}</span><span class='task-content index'><i class='fa fa-ellipsis-v'></i></span>`;
+    taskContainer.innerHTML = `<input class='task-content' type='checkbox' data-check><span class='task-content description'>${taskDetails.item}</span>
+    <input class='edit hide' value=${taskDetails.item}>
+    <span class='task-content index'></i></span>`;
+
+    const editBtn = document.createElement('i');
+    editBtn.className = 'fa';
+    editBtn.classList.add('fa-pencil');
+    taskContainer.appendChild(editBtn);
+
+    const removeBtn = document.createElement('i');
+    removeBtn.className = 'fa';
+    removeBtn.classList.add('fa-trash');
+    removeBtn.addEventListener('click', removeList);
+
+    taskContainer.appendChild(removeBtn);
     placeholder.appendChild(taskContainer);
+
+    editBtn.addEventListener('click', editDescription);
   }
-  return placeholder;
+};
+
+form.addEventListener('submit', () => {
+  addNewTask(simpleTodoTasks, input.value);
+  populateEachTask(simpleTodoTasks);
+  input.value = '';
+});
+
+const removeList = (e) => {
+  const li = e.target.closest('li');
+  const id = li.getAttribute('data-id');
+  removeTask(simpleTodoTasks, id);
+  simpleTodoTasks.forEach((obj, id) => { obj.index = id; });
+  populateEachTask(simpleTodoTasks);
+};
```

Comment:
- [ ] Ouch! this is close but it seems like the updated index values aren't reflected on the local storage in real-time and a page reload is required to update the local storage. After a few investigations, I noticed the problem is from the  `populateEachTask()` function. Check the below 👇
- On line `11` you're passing the wrong arguments to the `stringfy()` method
    ```
    localStorage.setItem('task', JSON.stringify(simpleTodoTasks)); ❌
    ```
- It should instead be like this:
    ```
    localStorage.setItem('task', JSON.stringify(arr)); 
    ```